okhttp-client-handler/monix/src/main/scala/com/softwaremill/sttp/okhttp/monix/OkHttpMonixClientHandler.scala

```diff
@@ -0,0 +1,79 @@
+package com.softwaremill.sttp.okhttp.monix
+
+import java.nio.ByteBuffer
+
+import com.softwaremill.sttp._
+import com.softwaremill.sttp.okhttp.OkHttpAsyncClientHandler
+import monix.eval.Task
+import monix.execution.{Cancelable, Scheduler}
+import monix.reactive.{Consumer, Observable}
+import okhttp3.{MediaType, OkHttpClient, RequestBody => OkHttpRequestBody}
+import okio.BufferedSink
+
+import scala.concurrent.Await
+import scala.concurrent.duration.Duration
+import scala.language.higherKinds
+import scala.util.{Failure, Success, Try}
+
+/**
+  * Created by omainegra on 8/4/17.
+  */
+class OkHttpMonixClientHandler private (client: OkHttpClient)(
+    implicit s: Scheduler)
+    extends OkHttpAsyncClientHandler[Task, Observable[ByteBuffer]](client,
+                                                                   TaskMonad) {
+
+  private lazy val io = Scheduler.io("sttp-monix-io")
+
+  override def streamToRequestBody(
+      stream: Observable[ByteBuffer]): Option[OkHttpRequestBody] =
+    Some(new OkHttpRequestBody() {
+      override def writeTo(sink: BufferedSink): Unit = {
+        val f = stream
+          .consumeWith(
+            Consumer.foreach(chunk => sink.write(chunk.array()))
+          )
+          .runAsync(io)
+
+        // We could safely block until the observable is consumed because OkHttp execute
+        // this method asynchronous in another ThreadPool.
+        Await.ready(f, Duration.Inf)
```

Comment:
I improve this part with suggestion from Alexandru in Monix gitter channel. The tip was convert the `Observable` to an `Iterable`. About the blocking part that's not correct. The thread that call `send` is never blocked because `OkHttp` internally run that code in another thread pool. Please check this example I've prepared.

```scala
// Start HttpServer
implicit val actorSystem: ActorSystem = ActorSystem()
implicit val materializer = ActorMaterializer()

val serverRoutes: Route =
  path("echo") {
    post {
      entity(as[String])(body => {
        complete(body)
      })
    }
  }

val endpoint = uri"http://localhost:31532"
Await.ready(
  Http().bindAndHandle(serverRoutes, endpoint.host, endpoint.port.get),
  Duration.Inf)

// Run Demo
implicit val scheduler = Scheduler.io("io")
implicit val handler = OkHttpMonixClientHandler()

val body = "sample string"
val observable = Observable
  .fromIterable(body.getBytes("utf-8"))
  .map(byte => ByteBuffer.wrap(Array(byte)))
  .delayOnNext(1.second)
  .doOnNext(byte => println2(s"Item produced: $byte"))

val request = sttp.post(uri"$endpoint/echo").streamBody(observable)
val response = request.response(asString).send().runAsync

while (!response.isCompleted) {
  println2(s"Still waiting ...")
  Thread.sleep(1000)
}
println2(s"Got response: ${response.value.get.get.body}")

Await.ready(actorSystem.terminate(), Duration.Inf)
assert(response.value.get.get.body == body)
```